### ISSUE TITLE
Invalid IDE0004 diagnostic

### ISSUE THREAD

**Issue by DomLatr** [dotnet/roslyn#40414]
**Version Used**: 

Visual Studio 16.4.1

**Steps to Reproduce**:

```
private long Repro()
{
    var random = new Random();
    long result = random.Next();
    result <<= 32;
    result |= (long)random.Next(); // IDE0004: Cast is redondant
    return result;
}
```

**Actual Behavior**:

After applying the fix, a new warning is reported `CS0675: Bitwise-or operator used on a sign-extended operand; consider casting to a smaller unsigned type first`

**Comment by CyrusNajmabadi** [comment]
This is an interesting case because the emitted-code/semantics are identical before/after. So the fix correctly determined this was safe and behavior preserving.  But it just doesn't know that the compiler more aggressively warns here.  We'll likely have to special case this.
